db/seeds.rb

```diff
@@ -156,17 +104,84 @@
 
 puts "They decided"
 
-puts "Create the Rulebooks..."
+puts "Creating the Rulebooks..."
 
 BOOK_TITLES.each_with_index do |book, index|
   genre = Genre.find_by(name: book[:genre])
   rulebook = Rulebook.create!(
     name: book[:title],
     description: book[:description],
-    genre: genre,
     img_url: book[:img]
     )
+  Pairing.create!(
+    genre: genre,
+    rulebook: rulebook
+    )
   puts "#{index + 1}. #{rulebook.name} published"
 end
 
-puts "All Rulesbooks created"
+puts "All Rulebooks created"
+
+puts "Generating Groups..."
+group_num = 0
+
+2.times do
+  preference = gandalf.preferences.sample.genre
+  rulebook = Rulebook.all.sample
+  group = Group.create!(
+    name: Faker::App.name,
+    description: Faker::Company.bs,
+    train_station: "Shibuya Station, Tokyo",
+    rulebook: rulebook,
+    gm: gandalf,
+    session_times: gandalf.availability
+    )
+  puts "#{group_num += 1}. #{group.name} created"
+end
+
+20.times do
+  area = data.sample
+  rulebook = Rulebook.all.sample
+  user = User.all.sample
+  status = true
+  while status
+    name = Faker::App.name
+    if !Group.find_by(name: name)
+      group = Group.create!(
+        name: name,
+        description: Faker::Company.bs,
+        rulebook: rulebook,
+        train_station: user.train_station,
+        gm: user,
+        session_times: user.availability
+        )
+      status = false
+      puts "#{group_num += 1}. #{group.name} - #{group.train_station} created"
+    end
+  end
+end
+
+puts "All Groups Created"
+
+puts "Linking Users to Groups..."
+groups = Group.where(train_station: 'Shibuya Station, Tokyo')
+groups.each do |group|
+  PlayerGroup.create!(user: gandalf, group: group) if group.gm != gandalf
+end
+
+30.times do
+  status = true
+  while status
+    user = User.all.sample
+    group = Group.all.sample
+    if group.gm != user && !group.users.include?(user) && group.users.count < 5
```

Comment:
maybe a condition we could add to check is whether the user.genres include the genre of the group's rulebook